dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo

```diff
@@ -148,24 +180,78 @@ partial model BaseTransformerVariableTap_INIT "Base model for initialization of
 
 equation
 
-  // Initial tap and ratio estimation
-  Tap0 = TapEstimation (ZPu, rTfoMinPu, rTfoMaxPu, NbTap, u10Pu, i10Pu, Uc20Pu);
+  // Initial ratio estimation
   if (NbTap == 1) then
     rTfo0Pu = rTfoMinPu;
   else
     rTfo0Pu = rTfoMinPu + (rTfoMaxPu - rTfoMinPu) * (Tap0 / (NbTap - 1));
   end if;
 
-  // Transformer equations
-  i10Pu = rTfo0Pu * (YPu * u20Pu - i20Pu);
-  rTfo0Pu * rTfo0Pu * u10Pu = rTfo0Pu * u20Pu + ZPu * i10Pu;
-
   // Voltage at terminal 2
   U20Pu = ComplexMath.'abs' (u20Pu);
 
 annotation(preferredView = "text");
-end BaseTransformerVariableTap_INIT;
+end BaseTransformerVariableTapCommon_INIT;
+
+partial model BaseIdealTransformerVariableTap_INIT "Base model for initialization of IdealTransformerVariableTap"
+  extends BaseTransformerVariableTapCommon_INIT;
+
+/* Equivalent circuit and conventions:
+
+               I1   r   I2
+    U1,P1,Q1 -->---oo---<-- U2,P2,Q2
+  (terminal1)              (terminal2)
+*/
+
+equation
+
+  // Initial tap estimation
+  Tap0 = IdealTransformerTapEstimation(rTfoMinPu, rTfoMaxPu, NbTap, u10Pu, Uc20Pu);
+
+  // Transformer equations
+  i10Pu = - rTfo0Pu * i20Pu;
+  rTfo0Pu * u10Pu = u20Pu;
+
+annotation(preferredView = "text");
+end BaseIdealTransformerVariableTap_INIT;
+
+partial model BaseNormalTransformerVariableTap_INIT "Base model for initialization of TransformerVariableTap"
```

Comment:
Remove "normal" from name?

Reply:
I wanted to keep similar names for the common part between ideal and classical transformers at initialization and time-domain stages. 
Thus I removed the normal here but added a Common in the base model name for the time-domain part so I have a symmetry between names in init and time-domain parts

Reply:
I don't see conflicts between base models so I would have expected the base model to be BaseTransformerVariableTap. You mention a symmetry with init models. If it is important to underline the symmetry I would have the "common" placed at the same place: BaseCommonBlabla or BaseBlaBlaCommon for both. But as their are all partial and won't be used as they are I don't think the symmetry is mandatory. So I let you decide between BaseTransformerVariableTap or BaseCommonTransformerVariableTap and BaseCommonTransformerVariableTap_INIT. What do you think?

Reply:
I don't really know. This is not big deal in any case: thus I came back to the version without symmetry and without the Normal part in the name.